server/models/User.js

```diff
@@ -19,7 +19,45 @@ const UserSchema = new Schema({
     date: {
         type: Date,
         default: Date.now
-    }
+    },
+    favorites: [
+        {
+            type: Schema.Types.ObjectId,
+            ref: "coffeeShops"
+        }
+    ]
 });
 
+UserSchema.statics.addFavorite = (userId, coffeeShopId) => {
+    const User = mongoose.model("users");
+    const CoffeeShop = mongoose.model("coffeeShops");
+
+    return User.findById(userId).then(user => {
+        return CoffeeShop.findById(coffeeShopId).then(shop => {
+            user.favorites.push(shop);
+            shop.users.push(user);
+
+            return Promise.all([user.save(), shop.save()]).then(
+                ([user, shop]) => user
+            );
+        });
+    });
+};
+
+UserSchema.statics.removeFavorite = (userId, coffeeShopId) => {
+    const User = mongoose.model("users");
+    const CoffeeShop = mongoose.model("coffeeShops");
+
+    return User.findById(userId).then(user => {
+        return CoffeeShop.findById(coffeeShopId).then(shop => {
```

Comment:
Should we add an if conditional here to make sure this user has this given coffeeShop?

Reply:
If there is not a coffeeshop associated with that user, then it will return an empty array if you query it.